extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -35,6 +35,46 @@ const BUILT_IN_FUNCTIONS = 'built-in-functions';
  * @const @private {!Object<string, !Object<string, Function>>}
  */
 const FUNCTION_WHITELIST = (function() {
+
+  /**
+   * Custom wrappers for mutating array methods so that we can offer the
+   * same functionality without allowing mutation on variables in bind scope.
+   */
+  const BindArrays = {
+    /**
+     * Similar to Array.prototype.splice, except it returns a copy of the
+     * passed-in array with the desired modifications.
+     * @param {!Array} array
+     * @param {number=} start
+     * @param {number=} end
+     * @param {...?} items
+     */
+    /*eslint "no-unused-vars": 0*/
+    'arraySplice': function(array, start, end, items) {
+      const copy = array.slice();
```

Comment:
We should verify that this is only called on array objects. 

Is there a good way to get this work as an array method? I.e. `someArray.copyAndSplice(...)`. The type-check logic would already exist then.